pbs/pbsrequest.go

```diff
@@ -30,20 +37,42 @@ type Bids struct {
 	Params     json.RawMessage `json:"params"`
 }
 
+type PBSVideo struct {
+	Mimes          []string `json:"mimes,omitempty"`
+	Minduration    uint64   `json:"minduration,omitempty"`
+	Maxduration    uint64   `json:"maxduration,omitempty"`
+	Startdelay     int64    `json:"startdelay,omitempty"`
+	Skippable      int      `json:"skippable,omitempty"`
+	PlaybackMethod int8     `json:"playback_method,omitempty"`
+	Frameworks     []string `json:"frameworks,omitempty"`
+}
+
 type AdUnit struct {
-	Code     string           `json:"code"`
-	TopFrame int8             `json:"is_top_frame"`
-	Sizes    []openrtb.Format `json:"sizes"`
-	Bids     []Bids           `json:"bids"`
-	ConfigID string           `json:"config_id"`
+	Code       string           `json:"code"`
+	TopFrame   int8             `json:"is_top_frame"`
+	Sizes      []openrtb.Format `json:"sizes"`
+	Bids       []Bids           `json:"bids"`
+	ConfigID   string           `json:"config_id"`
+	MediaTypes []string         `json:"media_types"`
 }
 
 type PBSAdUnit struct {
-	Sizes    []openrtb.Format
-	TopFrame int8
-	Code     string
-	BidID    string
-	Params   json.RawMessage
+	Sizes      []openrtb.Format
+	TopFrame   int8
+	Code       string
+	BidID      string
+	Params     json.RawMessage
+	Video      PBSVideo
+	MediaTypes []MediaType
```

Comment:
Agree. So, in this case:
- According to the spec, a Video object with empty `mimes` field is considered to be invalid;
- Ad unit with VIDEO present in media types but with missing or invalid Video object will be also considered invalid, and an attempt to convert it to OpenRTB request will cause an error.

I will implement this logic. Let me know if you think something should be added or changed.